module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -177,28 +175,48 @@ protected function findBestMatch(\VuFindSearch\Query\QueryInterface $query)
         }
 
         // // Find an exact match
-        // if (!array_key_exists($queryString, $this->subjectToId)) {
+        // if (!array_key_exists($queryString, $subjectToId)) {
         //     return false;
         // }
-        // $id = $this->subjectToId[$queryString];
+        // $id = $subjectToId[$queryString];
         // if (!$id) {
         //     return false;
         // }
 
-        $account = $this->idToAccount[$id];
+        $account = $idToAccount[$id];
         if (!$account) {
             return false;
         }
 
         return $account;
     }
 
+    /**
+     * Load or retrieve from the cache the list of LibGuides accounts
+     * from the LibGuides API.
+     *
+     * @return array An array containing the idToAccount and subjectToId maps
+     */
+    protected function loadDataIfNeeded()
```

Comment:
Nitpick, but I don't really like the method name `loadDataIfNeeded` -- that describes an internal implementation detail that the caller doesn't necessarily need to worry about. Maybe I'd call this one getLibGuidesData, since that's the one the user definitely wants to call, and then rename loadData to populateLibGuidesCache or something like that, to make it clear this is the primary entry point, and that is support for this.

Reply:
Thank you, I hated those two names and tried various options but couldn't come up with something reasonable.  I actually had "maybeLoadData()" at one point.  Your suggestions are much better.